### PR TITLE
Take version suffix into account for label

### PR DESCRIPTION
**Pull request by icecrime**
Signed-off-by: Arnaud Porterie <arnaud.porterie@docker.com>

### CODE REVIEW

**Comment by icecrime** [comment]
Ping @thaJeztah @cpuguy83 :innocent:

File: github/issue.go [hunk 1]
```diff
@@ -111,15 +111,31 @@ func (g GitHub) maybeOpinion(issueHook *octokat.IssueHook) error {
 	return nil
 }
 
+func labelFromVersion(version, suffix string) string {
+	switch suffix {
+	// Dev suffix is associated with a master build.
+	case "dev":
+		return "version/master"
```
Comment (thaJeztah) [inline github/issue.go]: Wonder if we should still add a version label -in addition- to this label. Allows for easier cleaning up "old" master issues

Comment (icecrime) [inline github/issue.go]: Isn't that just the creation date?

Comment (thaJeztah) [inline github/issue.go]: yes, that'd work, question is; is this label useful then, if it's not really used for filtering :smile:

Comment (icecrime) [inline github/issue.go]: There's no other way to learn about issues using a master build... If you exclude all `version/*` labels manually (which is cumbersome), you still have all the issues which are not providing a version number.

Comment (thaJeztah) [inline github/issue.go]: Hence me thinking to possibly add both (version + "master" / "dev" label). Perhaps I'm thinking too much, so I'm ok with merging as-is

File: github/issue.go [hunk 2]
```diff
@@ -111,15 +111,31 @@ func (g GitHub) maybeOpinion(issueHook *octokat.IssueHook) error {
...
+		return "version/" + version[0:strings.LastIndex(version, ".")]
+	// The default for unknown suffix is to consider the version unsupported.
+	default:
+		return "version/unsupported"
+	}
+}
+
 func (g GitHub) IssueAddVersionLabel(issueHook *octokat.IssueHook) error {
-	serverVersion := regexp.MustCompile(`Server:\s+Version:\s+(\d+\.\d+\.\d+)`)
+	serverVersion := regexp.MustCompile(`Server:\s+Version:\s+(\d+\.\d+\.\d+)-?(\S*)`)
```
Comment (cpuguy83) [inline github/issue.go]: What string is this matching? Doesn't seem to match the server version output from what I can see.

Comment (icecrime) [inline github/issue.go]: Why not?

Comment (cpuguy83) [inline github/issue.go]: Ok, this is working, I don't know what I did yesterday, mixed something up here.
Verified here: https://play.golang.org/p/ftkFyw_4Yy

**Comment by cpuguy83** [comment]
LGTM

**Comment by icecrime** [comment]
Thanks! I'll merge this, we'll probably refine with experience :+1:

**Pull request merged**